### PR TITLE
Add linting rules to editor config and check formatting in circle ci 

### PR DESCRIPTION
**Pull request by elcorbs**
I haven't run the linter because I didn't want to obscure the actual changes with linting changes for the review. So the build is failing on circle CI but I will run the linter in before merging.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -159,6 +162,19 @@ jobs:
...
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Install dotnet format
+          command: dotnet tool install -g dotnet-format
+      - run:
+          name: Run formatter check
+          command: dotnet format --dry-run --check
+          environment:
+            PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/.dotnet/tools
```
Comment (Renny4Real) [inline .circleci/config.yml:177]: Should we save this in Circle CI as an environment variable? Looks pretty long 😄

File: .editorconfig [hunk 2]
```diff
@@ -7,16 +7,116 @@ indent_style=space
...
+csharp_space_after_semicolon_in_for_statement = true
+csharp_space_before_semicolon_in_for_statement = false
+csharp_space_around_declaration_statements = false
+csharp_space_before_open_square_brackets = false
+csharp_space_between_empty_square_brackets = false
+csharp_space_between_square_brackets = false
+
+#Using directives
+csharp_using_directive_placement = outside_namespace
+
```
Comment (Renny4Real) [inline .editorconfig:122]: I might have missed this but are we removing unused directives?

Comment (elcorbs) [reply .editorconfig:122]: I couldn't find a setting to specify this, unfortunately :(

Comment (Renny4Real) [reply .editorconfig:122]: 👍

File: .circleci/config.yml (hunk 1 above)
Comment (elcorbs) [reply .circleci/config.yml]: I've removed this variable and installed the tool to a path instead :)

**Pull request merged**